contrib/babelfishpg_tsql/src/pltsql_coerce.c

```diff
@@ -1288,6 +1348,41 @@ validate_special_function(char *func_nsname, char *func_name, List* fargs, int n
 	return true;
 }
 
+/*
+ * Selects the best candidate function for HASHBYTES based on input types.
+ * Focuses on matching the second argument type, particularly for NVARCHAR.
+ * Returns a single best candidate from the sys schema, or NULL if none found.
+ */
+static FuncCandidateList
+tsql_func_select_candidate_for_hashbytes(Oid *input_typeids, FuncCandidateList candidates)
+{
+
+    FuncCandidateList        current_candidate, best_candidate;
+    Oid                sys_oid = get_namespace_oid("sys", false);
+    Oid                 *argtypes;
+    int                nargs_func = 0;
+
+    /* Get the candidate with matching second argument type */
+    best_candidate = NULL;
+    for (current_candidate = candidates;
+        current_candidate != NULL;
+        current_candidate = current_candidate->next)
+    {
+        /* we should only consider candidates for hashbytes function from sys schema */
+        if (get_func_namespace(current_candidate->oid) != sys_oid)
+            return NULL;
```

Comment:
there could be sys.hashbytes available in next iteration

Reply:
But since user has defined their own function hashbytes, so we don't need to look/wait for any sys.hashbytes function.